File: Bpa_v2_bakc/src/main/java/com/example/Bpa_v2_bakc/entities/sql_server/Detail.java
package com.example.Bpa_v2_bakc.entities.sql_server;

import com.example.Bpa_v2_bakc.entities.mysql.Etat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "detail")
public class Detail {
    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_detail")
    private int idDetail;

    @Column(unique=true)
    private String uid;
    private String ref_dem;
    private String code_article;
    private String quantite;
    private String pu;
    private String objet;
    private String montant;
    private String fournisseur;
    private String devise;
    private String id_x3;

    @ManyToOne
    @JoinColumn(name = "etat", referencedColumnName = "id_etat")
    private Etat etat;
}

File: Bpa_v2_front/src/app/models/detail.ts
import { Etat } from "./etat";

export interface Detail{
    idDetail: number;
    ref_dem: string;
    uid: string;
    code_article: string;
    quantite: string;
    pu: string;
    objet: string;
    montant: string;
    devise: string;
    fournisseur: string;
    id_x3: string;
    etat: Etat;
} 